lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +132,48 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+*/
+
+function getSolutionsAndSymbol (equation) {
+  // Gets all factors that aren't a constant
+  const factors = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
+
+  const solutions = [];
+
+  let symbol;
+
+  // For each factor, solve the equation factor = 0 for the symbol
+  for (var f in factors) {
+    const leftNode = Node.Type.isParenthesis(factors[f])
+          ? factors[f].content
+          : factors[f];
+
+    const rightNode = Node.Creator.constant(0);
+
+    const allSteps = stepThrough(leftNode, rightNode, '=');
+
+    if (allSteps.length > 0) {
+      const lastStep = allSteps.slice(-1)[0];
+
+      symbol = lastStep.newEquation.leftNode;
+      // Append to a list of roots
+      solutions.push(lastStep.newEquation.rightNode);
+    }
+    else {
```

Comment:
I really don't like that you assume 0 is a solution if there are no steps - can you explicitly check for that, and otherwise don't push anything to `solutions`, and check at the end to make sure solutions has length > 0 (and if not, return something that shows nothing was found or changed) - that'd make me feel a lot safer about all this, that we're handling missing cases